beacon-chain/blockchain/service.go

```diff
@@ -580,3 +580,50 @@ func spawnCountdownIfPreGenesis(ctx context.Context, genesisTime time.Time, db d
 	}
 	go slots.CountdownToGenesis(ctx, genesisTime, uint64(gState.NumValidators()), gRoot)
 }
+
+func (s *Service) detectEquivocatingBlock(ctx context.Context, block interfaces.ReadOnlySignedBeaconBlock) (*ethpb.ProposerSlashing, error) {
+	// Get the incoming block's header
+	header1, err := block.Header()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get header from incoming block")
+	}
+
+	s.headLock.RLock()
+	headBlock, err := s.headBlock()
+	if err != nil {
+		s.headLock.RUnlock()
+		return nil, errors.Wrap(err, "could not get head block")
+	}
+	s.headLock.RUnlock()
+
+	// Get the head block's header
+	header2, err := headBlock.Header()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get header from head block")
+	}
+
+	// Check for equivocation:
+	if header1.Header.Slot == header2.Header.Slot &&
+		header1.Header.ProposerIndex == header2.Header.ProposerIndex {
+
+		header1Root, err := header1.Header.HashTreeRoot()
```

Comment:
No need to hash again here, you already have the root when you call this helper.

Reply:
fixed this in the next commit